RELEASING.md

```diff
@@ -1,48 +1,63 @@
-# Releasing OpenTelemetry Packages (for maintainers only)
-This document explains how to publish all OT modules at version x.y.z. Ensure that you’re following semver when choosing a version number.
+# Release instructions
+
+## Preparing a new major or minor release
+
+* Merge a pull request to `main` updating the `CHANGELOG.md`.
+  * The heading for the release should include the release version but not the release date, e.g.
+    `## Version 1.9.0 (unreleased)`.
+  * Use `.github/scripts/draft-change-log-entries.sh` as a starting point for writing the change
+    log.
+* Run the [Prepare release branch workflow](https://github.com/trask/repository-template/actions/workflows/prepare-release-branch.yml).
+  * Press the "Run workflow" button, and leave the default branch `main` selected.
+* Review and merge the two pull requests that it creates
+  (one is targeted to the release branch and one is targeted to `main`).
+
+## Preparing a new patch release
+
+* Backport pull request(s) to the release branch.
```

Comment:
Under which circumstances do we do a patch release? Is this the same as "hotfix" procedure down below?

Reply:
Patch release is any release that bumps the last digit, e.g. `1.10.1` is a patch release on top of `1.10.0`. I believe this is synonymous with hotfix.

Although it looks like your hotfix procedure is currently to only publish the affected packages:

> With administrator priviledges for PyPi, manually publish the affected packages

In Java we do a full publishing of all packages since that is simpler for us and works with the same automation as a regular release.

Reply:
I (optimistically) removed the "Hotfix procedure" section

Reply:
Hmm, it could be debated on whether we want to do a release of all packages if the hotfix only pertains to a few. Maybe we can do a separated "hotfix" release process in which we can specify which packages to release?

Reply:
> Hmm, it could be debated on whether we want to do a release of all packages if the hotfix only pertains to a few.

I personally think it's more effort to do a targeted release of only a few packages, plus by always releasing all packages at the same versions, it's typically easier (at least in the Java world) for users to target all of their dependencies to the same matching version.

But I'm happy to bring back the hotfix release process section, just let me know.

Reply:
Not a huge opinion from my side. I'm happy to test out this way for now and leave out the hotfix specific section.